packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -892,7 +879,22 @@ function commitUnmount(
             const {destroy, tag} = effect;
             if (destroy !== undefined) {
               if ((tag & HookPassive) !== NoHookEffect) {
-                enqueuePendingPassiveHookEffectUnmount(current, effect);
+                effect.tag |= HookHasEffect;
+
+                current.effectTag |= Passive;
```

Comment:
Trying to move away from mutating the current tree.

I think we can move these checks to passive phase after paint?

Reply:
Hm, maybe. We'd need to track when we're inside of an unmounted tree, so we would know to always run passive effects (regardless of `HookHasEffect`).

Reply:
I'm going to leave this ^ to the TODO as well.